src/NServiceBus.Transport.SQS/InputQueuePump.cs

```diff
@@ -415,9 +415,24 @@ public static TransportMessage ExtractTransportMessage(Message receivedMessage,
             }
             // HINT: Message Id is the only required header
             transportMessage.Headers.TryAdd(Headers.MessageId, messageIdOverride);
+            AddCustomNativeHeadersToNServiceBusHeaders(receivedMessage, transportMessage);
+
             return transportMessage;
         }
 
+        static void AddCustomNativeHeadersToNServiceBusHeaders(Message receivedMessage, TransportMessage transportMessage)
+        {
+            foreach (var receivedMessageMessageAttribute in receivedMessage.MessageAttributes)
+            {
+                // TODO is there a better way to achieve the same result? This seems fragile
+                if (TransportHeaders.AllTransportHeaders.Contains(receivedMessageMessageAttribute.Key) || receivedMessageMessageAttribute.Key == Headers.MessageId)
+                {
+                    continue;
+                }
+                _ = transportMessage.Headers.TryAdd(receivedMessageMessageAttribute.Key, receivedMessageMessageAttribute.Value?.StringValue ?? string.Empty);
```

Comment:
`TryAdd` makes it so that NServiveBus headers will never be overridden. Should native headers be able to override NServiceBus headers?

Reply:
We have discussed this and concluded for native integration scenarios it makes sense for the message attributes to take precedence when there are duplicates. The likelihood of duplicates is very low anyway.